app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt

```diff
@@ -0,0 +1,81 @@
+package nerd.tuxmobil.fahrplan.congress
+
+import android.app.Application
+import android.util.Log
+
+import androidx.annotation.CallSuper
+
+import org.ligi.tracedroid.TraceDroid
+
+import java.util.*
+
+import info.metadude.android.eventfahrplan.commons.logging.Logging.Companion.get
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository.initialize
+import nerd.tuxmobil.fahrplan.congress.utils.ConferenceTimeFrame
+
+class MyApp : Application() {
+
+    enum class TASKS {
+        NONE, FETCH, PARSE, FETCH_CANCELLED
+    }
+
+    @CallSuper
+    override fun onCreate() {
+        super.onCreate()
+        TraceDroid.init(this)
+        task_running = TASKS.NONE
+        initialize(
+            applicationContext,
+            get()
+        )
+    }
+
+    companion object {
+
+        const val DEBUG = false
+
+        private val FIRST_DAY_START = getMilliseconds(
+            "Europe/Paris",
+            BuildConfig.SCHEDULE_FIRST_DAY_START_YEAR,
+            BuildConfig.SCHEDULE_FIRST_DAY_START_MONTH,
+            BuildConfig.SCHEDULE_FIRST_DAY_START_DAY
+        )
+
+        private val LAST_DAY_END = getMilliseconds(
+            "Europe/Paris",
+            BuildConfig.SCHEDULE_LAST_DAY_END_YEAR,
+            BuildConfig.SCHEDULE_LAST_DAY_END_MONTH,
+            BuildConfig.SCHEDULE_LAST_DAY_END_DAY
+        )
+
+        @JvmField
+        val conferenceTimeFrame = ConferenceTimeFrame(FIRST_DAY_START, LAST_DAY_END)
+
+        @JvmField
+        var task_running = TASKS.NONE
+
+        private fun getMilliseconds(
```

Comment:
A bit more concise:

``` kotlin
private fun getMilliseconds(timeZoneId: String, year: Int, month: Int, day: Int): Long {
    val zeroBasedMonth = month - 1
    val zone = TimeZone.getTimeZone(timeZoneId)
    return GregorianCalendar(zone).apply {
        set(year, zeroBasedMonth, day, 0, 0, 0)
        set(Calendar.MILLISECOND, 0)
    }.timeInMillis
}
```

Please move the helper function under the properties where it is used.